Mathlib/Lean/Meta/Simp.lean

```diff
@@ -0,0 +1,36 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Lean
+
+/-!
+# Helper functions for using the simplifier.
+-/
+
+open Lean Elab.Tactic
+
+namespace Lean.Meta
+
+/-- Construct a `SimpTheorems` from a list of names. (i.e. as with `simp only`). -/
+def simpTheoremsOfNames (lemmas : List Name) : MetaM SimpTheorems := do
+  lemmas.foldlM (·.addConst ·) (← simpOnlyBuiltins.foldlM (·.addConst ·) {})
+
+/-- Simplify an expression using only a list of lemmas specified by name. -/
+def simpOnlyNames (lemmas : List Name) (e : Expr) : MetaM Simp.Result := do
```

Comment:
We need to write a `mkSimpContext` with all the bells an whistles of `simp` (and then maybe PR that to core).
Another partial implementation is in `Tactic.Simps.Basic` (that should be moved to this file).

Reply:
Thanks. I've left a comment pointing to that other partial implementation.